command/agent_test.go

```diff
@@ -2082,3 +2082,118 @@ listener "tcp" {
 		"Points",
 	})
 }
+
+func TestAgent_Quit(t *testing.T) {
+	//----------------------------------------------------
+	// Start the server and agent
+	//----------------------------------------------------
+	logger := logging.NewVaultLogger(hclog.Error)
+	cluster := vault.NewTestCluster(t,
+		&vault.CoreConfig{
+			Logger: logger,
+			CredentialBackends: map[string]logical.Factory{
+				"approle": credAppRole.Factory,
+			},
+			LogicalBackends: map[string]logical.Factory{
+				"kv": logicalKv.Factory,
+			},
+		},
+		&vault.TestClusterOptions{
+			NumCores: 1,
+		})
+	cluster.Start()
+	defer cluster.Cleanup()
```

Comment:
nit: lately I have been using `t.Cleanup()` instead of a defer, ideally `vault.NewTestCluster()` would take care of this for us.
```suggestion
	t.Cleanup(cluster.Cleanup)
```

Reply:
I actually prefer `defer` tbh - it's much more commonly understood and doesn't have any disadvantages in this instance from `t.Cleanup` AFAICT. Agreed it would be better if `vault.NewTestCluster()` did this for us though, that would be a nice improvement.